File: dotnet/src/CarespaceSDK/Api/ClientsApi.cs
using CarespaceSDK.Http;
using CarespaceSDK.Models;

namespace CarespaceSDK.Api;

public class ClientsApi : IClientsApi
{
    private readonly IHttpClient _httpClient;

    public ClientsApi(IHttpClient httpClient)
    {
        _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
    }

    public async Task<PaginatedResponse<Client>> GetClientsAsync(
        int page = 1, 
        int limit = 20, 
        string? search = null, 
        string? providerId = null, 
        bool? isActive = null,
        CancellationToken cancellationToken = default)
    {
        var queryParams = new List<string>();
        queryParams.Add($"page={page}");
        queryParams.Add($"limit={limit}");
        
        if (!string.IsNullOrEmpty(search))
            queryParams.Add($"search={Uri.EscapeDataString(search)}");
        
        if (!string.IsNullOrEmpty(providerId))
            queryParams.Add($"provider_id={providerId}");
        
        if (isActive.HasValue)
            queryParams.Add($"is_active={isActive.Value}");

        var query = string.Join("&", queryParams);
        var endpoint = $"/clients?{query}";

        return await _httpClient.GetAsync<PaginatedResponse<Client>>(endpoint, cancellationToken);
    }

    public async Task<ApiResponse<Client>> GetClientAsync(string clientId, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(clientId)) throw new ArgumentException("Client ID is required", nameof(clientId));
        return await _httpClient.GetAsync<ApiResponse<Client>>($"/clients/{clientId}", cancellationToken);
    }

    public async Task<ApiResponse<Client>> CreateClientAsync(CreateClientRequest request, CancellationToken cancellationToken = default)
    {
        if (request == null) throw new ArgumentNullException(nameof(request));
        return await _httpClient.PostAsync<ApiResponse<Client>>("/clients", request, cancellationToken);
    }

    public async Task<ApiResponse<Client>> UpdateClientAsync(string clientId, UpdateClientRequest request, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(clientId)) throw new ArgumentException("Client ID is required", nameof(clientId));
        if (request == null) throw new ArgumentNullException(nameof(request));
        return await _httpClient.PutAsync<ApiResponse<Client>>($"/clients/{clientId}", request, cancellationToken);
    }

    public async Task DeleteClientAsync(string clientId, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(clientId)) throw new ArgumentException("Client ID is required", nameof(clientId));
        await _httpClient.DeleteAsync($"/clients/{clientId}", cancellationToken);
    }

    public async Task<ApiResponse<ClientStats>> GetClientStatsAsync(string clientId, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(clientId)) throw new ArgumentException("Client ID is required", nameof(clientId));
        return await _httpClient.GetAsync<ApiResponse<ClientStats>>($"/clients/{clientId}/stats", cancellationToken);
    }

    public async Task<PaginatedResponse<Program>> GetClientProgramsAsync(string clientId, int page = 1, int limit = 20, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(clientId)) throw new ArgumentException("Client ID is required", nameof(clientId));
        
        var queryParams = new List<string>();
        queryParams.Add($"page={page}");
        queryParams.Add($"limit={limit}");
        var query = string.Join("&", queryParams);
        
        return await _httpClient.GetAsync<PaginatedResponse<Program>>($"/clients/{clientId}/programs?{query}", cancellationToken);
    }

    public async Task<ApiResponse<object>> AssignProgramAsync(string clientId, string programId, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(clientId)) throw new ArgumentException("Client ID is required", nameof(clientId));
        if (string.IsNullOrWhiteSpace(programId)) throw new ArgumentException("Program ID is required", nameof(programId));
        
        var request = new { program_id = programId };
        return await _httpClient.PostAsync<ApiResponse<object>>($"/clients/{clientId}/programs", request, cancellationToken);
    }

    public async Task<ApiResponse<object>> UnassignProgramAsync(string clientId, string programId, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(clientId)) throw new ArgumentException("Client ID is required", nameof(clientId));
        if (string.IsNullOrWhiteSpace(programId)) throw new ArgumentException("Program ID is required", nameof(programId));
        
        return await _httpClient.DeleteAsync<ApiResponse<object>>($"/clients/{clientId}/programs/{programId}", cancellationToken);
    }

    public async Task<ApiResponse<object>> DeactivateClientAsync(string clientId, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(clientId)) throw new ArgumentException("Client ID is required", nameof(clientId));
        return await _httpClient.PostAsync<ApiResponse<object>>($"/clients/{clientId}/deactivate", null, cancellationToken);
    }

    public async Task<ApiResponse<object>> ActivateClientAsync(string clientId, CancellationToken cancellationToken = default)
    {
        if (string.IsNullOrWhiteSpace(clientId)) throw new ArgumentException("Client ID is required", nameof(clientId));
        return await _httpClient.PostAsync<ApiResponse<object>>($"/clients/{clientId}/activate", null, cancellationToken);
    }
}

File: kotlin/carespace-sdk/src/commonMain/kotlin/com/carespace/sdk/api/ClientsApi.kt
package com.carespace.sdk.api

import com.carespace.sdk.http.CarespaceHttpClient
import com.carespace.sdk.models.*

/**
 * Clients API interface
 */
interface ClientsApi {
    suspend fun getClients(
        page: Int = 1,
        limit: Int = 20,
        search: String? = null,
        providerId: String? = null,
        isActive: Boolean? = null
    ): PaginatedResponse<Client>

    suspend fun getClient(clientId: String): ApiResponse<Client>
    suspend fun createClient(request: CreateClientRequest): ApiResponse<Client>
    suspend fun updateClient(clientId: String, request: UpdateClientRequest): ApiResponse<Client>
    suspend fun deleteClient(clientId: String)
    suspend fun getClientStats(clientId: String): ApiResponse<ClientStats>
    suspend fun getClientPrograms(clientId: String, page: Int = 1, limit: Int = 20): PaginatedResponse<Program>
    suspend fun assignProgram(clientId: String, programId: String): ApiResponse<Unit>
    suspend fun unassignProgram(clientId: String, programId: String): ApiResponse<Unit>
    suspend fun deactivateClient(clientId: String): ApiResponse<Unit>
    suspend fun activateClient(clientId: String): ApiResponse<Unit>
}

/**
 * Clients API implementation
 */
class ClientsApiImpl(private val httpClient: CarespaceHttpClient) : ClientsApi {

    override suspend fun getClients(
        page: Int,
        limit: Int,
        search: String?,
        providerId: String?,
        isActive: Boolean?
    ): PaginatedResponse<Client> {
        val parameters = buildMap {
            put("page", page)
            put("limit", limit)
            search?.let { put("search", it) }
            providerId?.let { put("provider_id", it) }
            isActive?.let { put("is_active", it) }
        }
        return httpClient.get("/clients", parameters)
    }

    override suspend fun getClient(clientId: String): ApiResponse<Client> {
        require(clientId.isNotBlank()) { "Client ID cannot be blank" }
        return httpClient.get("/clients/$clientId")
    }

    override suspend fun createClient(request: CreateClientRequest): ApiResponse<Client> {
        return httpClient.post("/clients", request)
    }

    override suspend fun updateClient(clientId: String, request: UpdateClientRequest): ApiResponse<Client> {
        require(clientId.isNotBlank()) { "Client ID cannot be blank" }
        return httpClient.put("/clients/$clientId", request)
    }

    override suspend fun deleteClient(clientId: String) {
        require(clientId.isNotBlank()) { "Client ID cannot be blank" }
        httpClient.delete("/clients/$clientId")
    }

    override suspend fun getClientStats(clientId: String): ApiResponse<ClientStats> {
        require(clientId.isNotBlank()) { "Client ID cannot be blank" }
        return httpClient.get("/clients/$clientId/stats")
    }

    override suspend fun getClientPrograms(clientId: String, page: Int, limit: Int): PaginatedResponse<Program> {
        require(clientId.isNotBlank()) { "Client ID cannot be blank" }
        val parameters = mapOf("page" to page, "limit" to limit)
        return httpClient.get("/clients/$clientId/programs", parameters)
    }

    override suspend fun assignProgram(clientId: String, programId: String): ApiResponse<Unit> {
        require(clientId.isNotBlank()) { "Client ID cannot be blank" }
        require(programId.isNotBlank()) { "Program ID cannot be blank" }
        val request = mapOf("program_id" to programId)
        return httpClient.post("/clients/$clientId/programs", request)
    }

    override suspend fun unassignProgram(clientId: String, programId: String): ApiResponse<Unit> {
        require(clientId.isNotBlank()) { "Client ID cannot be blank" }
        require(programId.isNotBlank()) { "Program ID cannot be blank" }
        return httpClient.delete("/clients/$clientId/programs/$programId")
    }

    override suspend fun deactivateClient(clientId: String): ApiResponse<Unit> {
        require(clientId.isNotBlank()) { "Client ID cannot be blank" }
        return httpClient.post("/clients/$clientId/deactivate")
    }

    override suspend fun activateClient(clientId: String): ApiResponse<Unit> {
        require(clientId.isNotBlank()) { "Client ID cannot be blank" }
        return httpClient.post("/clients/$clientId/activate")
    }
}